### PR TITLE
Add a configurable Access-Control-Max-Age header

### PR DESCRIPTION
**Pull request by miike**
This adds an additional header to allow caching of the OPTIONS response from the collector see #3904 

@BenFradet Where do you think I should put this property in the configuration hocon? I'm not sure that putting it at the root of the collector config is the best option.

### CODE REVIEW

**Comment by miike** [comment]
One of the tests is failing at the moment. I'll fix this once we've confirmed where we want to put the configuration option.

**Comment by snowplowcla** [comment]
@miike  has signed the Software Grant and Corporate Contributor License Agreement

File: 2-collectors/scala-stream-collector/config.hocon.sample [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  remote-address-header = on
+
+  uri-parsing-mode = relaxed
+  raw-request-uri-header = on
+
+  # Define the maximum request length (the default is 2048)
+  parsing {
+    max-uri-length = 32768
+  }
+}
```
Comment (BenFradet) [inline 2-collectors/scala-stream-collector/config.hocon.sample]: please modify the existing config: https://github.com/snowplow/snowplow/blob/master/2-collectors/scala-stream-collector/examples/config.hocon.sample

File: 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala [hunk 2]
```diff
@@ -113,6 +113,7 @@ package model {
   final case class CollectorConfig(
     interface: String,
     port: Int,
+    accessControlMaxAge: Int,
```
Comment (BenFradet) [inline 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala]: should this be optional?

**Comment by BenFradet** [comment]
hey @miike , could you update the PR?

**Comment by BenFradet** [comment]
Incorporated into #3965 , thanks a lot! :+1:

**Pull request closed without merging**